interview-questions.md

```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer:Both map and filter methods create a new array and do not change the original array. The map method performs the function once to every value in the array. The filter method only allows the values that pass its criteria to move into the new array.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is the repeat of a series of event until a specific outcome is achieved. For an example, we could run a for loop for the argument of Mississippi. What we want is the amount of times the letter "s" appears because who wants to count that out? We would be able to count it out with the for loop to loop through each letter in the word "Mississpi" through the index starting at 0 or M and ending at 10 or i. It would result in us finding the amount of "s's" in the word which is 4.
 
-Researched answer:
+Researched answer: A way to run a block of code until a certain condition is met. It can be refered to as looping as well. Other loops include the while, do while, for, for in, break and continue. They each have their own use in the iteration process.
 
 5. What is the difference between a class and an object?
 
-Your answer:
+Your answer: A class is a variable that is able to be used to declare the blueprints for an object. An can be a result of a class or be a variable that has functions, strings, symbols, and numbers contained within. For an example, we could make a class to define what makes a user profile online. Once the neccessary information is inputted, the result of the class is a new object with identifying properties as all people online are different.
 
-Researched answer:
+Researched answer: Classes are able to define how an object is made. They contain what pertains to what can be and what is the object. An object is something that has its own properties, values, and functions. The properties, values, and functions are created through name value pairs in the body of the object.
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: Hoisting in Javascript is the act of pulling the code through a full loop of maintenace. This is done by having a critical error in the system and having to make a copy of the code in another file to troubleshoot to address the issue while still having it run.
 
-Researched answer:
+Researched answer: I apologize for the previous answer. Hoisting is the act of allowing a variable to be pertaining to the function by bringing it up the block of code to be declared as the appropriate name. It is so the variable can be properly referenced and not result in an undefined.
```

Comment:
Really great work in this section.  All answers are really well thought out and researched.